fairdatapipeline/fdp_utils.py

```diff
@@ -105,7 +104,10 @@ def extract_id(url: str) -> str:
     Returns:
         |   str: id derrived from the url
     """
-    return list(filter(None, urlsplit(url).path.split("/")))[-1]
+    try:
+        return list(filter(None, urlsplit(url).path.split("/")))[-1]
+    except:
```

Comment:
Do not catch an exception without explicitly stating the exception type as this will catch all possible errors making it impossible to debug any other issues that may be occurring, if this exception is to cover the case where `[-1]` fails then use `except IndexError:`.  

Also if the intention here is to return `None` when there are no results for the URL split you do not need to use filter, as an empty list `[]` behaves as `None` in an `if` statement anyway, i.e. `if my_list:` would return `False` if the list is empty. From what I can see you should not need to catch the exception either.

Consider splitting this step into more steps to easily identify any problems, i.e.:
```python
split_url_path = urlsplit(url).path.split("/")

if not split_url_path:
    raise IndexError(f"Unable to extract ID from registry URL: {url}")
```

Reply:
I believe this was a refactor suggested by sourcery to have a more compact code.

does this solve it? 
```
    try:
        return list(filter(None, urlsplit(url).path.split("/")))[-1]
    except IndexError:
        raise IndexError(f"Unable to index input url: {url}")
```

Reply:
I do not see why `filter` is needed at all

Reply:
ok using your idea

```
    split_url_path = urlsplit(url).path.split("/")
    if not split_url_path:
        raise IndexError(f"Unable to extract ID from registry URL: {url}")
    return [s for s in split_url_path if s != ''][-1]
```